### PR TITLE
bpo-39664: Add tests for operator module.

### PR DESCRIPTION
**Pull request by tirkarthi**
Add tests for operator module to improve coverage.

* iconcat should have __getitem__ implemented.
* Test index and not functions.
* Test length_hint default value type.

https://bugs.python.org/issue39664

### CODE REVIEW

File: Lib/test/test_operator.py [hunk 1]
```diff
@@ -474,6 +474,30 @@ def __getitem__(self, other): return 5  # so that C is a sequence
         self.assertEqual(operator.ixor     (c, 5), "ixor")
         self.assertEqual(operator.iconcat  (c, c), "iadd")
 
+    def test_iconcat_without_getitem(self):
+        operator = self.module
+        class X: pass
+
+        msg = "'X' object can't be concatenated"
+        with self.assertRaisesRegex(TypeError, msg):
+            operator.iconcat(X(), X())
```
Comment (serhiy-storchaka) [inline Lib/test/test_operator.py]: I think it is better to test some objects which support addition, for example
```suggestion
            operator.iconcat(1, 0.5)
```

File: Lib/test/test_operator.py [hunk 2]
```diff
@@ -474,6 +474,30 @@ def __getitem__(self, other): return 5  # so that C is a sequence
...
+
+    def test_index(self):
```
Comment (serhiy-storchaka) [inline Lib/test/test_operator.py:484]: `operator.index` is already tested in `Lib/test/test_index.py`.

File: Lib/test/test_operator.py [hunk 3]
```diff
@@ -474,6 +474,30 @@ def __getitem__(self, other): return 5  # so that C is a sequence
...
+        operator = self.module
+        class X:
+            def __index__(self):
+                return 1
+
+        self.assertEqual(operator.index(X()), 1)
+
+    def test_not_(self):
```
Comment (serhiy-storchaka) [inline Lib/test/test_operator.py:492]: You can take `test_truth` as a pattern for `test_not_`.

File: Lib/test/test_operator.py (hunk 2 above)
Comment (tirkarthi) [reply Lib/test/test_operator.py:484]: I guess test_index only checks the C version of the module. This test covers the Python version with `self.module` used for C and Python version of the module.

Comment (serhiy-storchaka) [reply Lib/test/test_operator.py:484]: Agree. Although it would be nice to extend this test. At least test it with int, float, Fraction, Decimal, None.

**Review by iritkatriel — CHANGES_REQUESTED** [review]
There is a merge conflict.

**Comment by iritkatriel** [comment]
Marking as pending because this has been stale for a couple of years with no followup to the code review.  If this is abandoned we should perhaps close it.

**Comment by tirkarthi** [comment]
Thanks @iritkatriel for triaging. I am okay with closing this since this was mostly to make 100% code coverage and I am not actively working on this.

**Comment by hugovk** [comment]
I attempted to update this, but it won't let me push to it, so will make a new PR instead.

---

For reference:

```console
$ gh co 18537 # branch is add-test-operator-module:refs/pull/18537/head
$ git merge upstream/main
$ # resolve conflict
$ git push
fatal: The upstream branch of your current branch does not match
the name of your current branch.  To push to the upstream branch
on the remote, use

    git push upstream HEAD:refs/pull/18537/head

To push to the branch of the same name on the remote, use

    git push upstream HEAD

To choose either option permanently, see push.default in 'git help config'.

To avoid automatically configuring an upstream branch when its name
won't match the local branch, see option 'simple' of branch.autoSetupMerge
in 'git help config'.
$ git push upstream HEAD:refs/pull/18537/head
Found existing alias for "git push". You should use: "gp"
Enumerating objects: 23, done.
Counting objects: 100% (23/23), done.
Delta compression using up to 8 threads
Compressing objects: 100% (15/15), done.
Writing objects: 100% (15/15), 1.81 KiB | 1.81 MiB/s, done.
Total 15 (delta 12), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (12/12), completed with 7 local objects.
To https://github.com/python/cpython.git
 ! [remote rejected]       HEAD -> refs/pull/18537/head (deny updating a hidden ref)
```

**Comment by hugovk** [comment]
Please see PR GH-115883.

**Pull request closed without merging**